src/mon/PGMonitor.cc

```diff
@@ -2083,7 +2083,7 @@ int PGMonitor::_warn_slow_request_histogram(const pow2_hist_t& h, string suffix,
                                             list<pair<health_status_t,string> > *detail) const
 {
   unsigned sum = 0;
-  for (unsigned i = h.h.size() - 1; i > 0; --i) {
+  for (int i = h.h.size() - 1; i > 0; --i) {
```

Comment:
this is wrong. if `h.h.empty()`, `(h.h.size() - 1)` will underflow, because `h.h.size()` is an unsigned integer. better off returning `0` right away, if `h.h.empty()`.